src/keys/secret.rs

```diff
@@ -462,3 +471,177 @@ impl SecretKeyVarGen {
         SignatureVarGen::new(u, R)
     }
 }
+
+/// Implementation of the `SpeedyMuSig` Schnorr-based
+/// multisignature scheme. It allows several signers to
+/// create a signature that proves a message to be signed by
+/// them all, given their public keys. The signature can be
+/// verified using the same function used for the standard
+/// Schnorr signature, using the sum of all the signers' public keys.
+///
+/// reference: https://eprint.iacr.org/2021/1375.pdf - pag. 19
+///
+/// # Feature
+///
+/// Only available with the "multisig" feature enabled.
+#[cfg(feature = "multisig")]
+#[allow(non_snake_case)]
+pub trait SecretKeyMultisig {
```

Comment:
Is there a reason why you add this as a trait and not implement the methods directly on `SecretKey`? Will we want to implement this trait on other structs in the future?

Reply:
the only reason is that I wanted somehow to isolate the code to avoid having to add the macros so many times and also have a single `///` comment describing the full module (trait). But now that you mention it, yes, maybe some day we could have a different kind of struct implementing it

Reply:
The reason I asked was that I didn't really see why this is a trait since the only reason to work with traits is to implement the same functionalities on different types, which is a functionality we don't really need.

That's why in this case I actually don't think a trait is a good choice for implementing the multisig behavior.

I would not make the functions methods on the `SecretKey` but simply functions in themselves. In that case you can also put them into a separate module at crate root level where the functions are namespaces by the module name:

```rust
use jubjub_schnorr::multisig;

multisig::round_1(rng);
multisig::round_2(sk, r, s, pk_vec, R_vec, S_vec, msg);
multisig::combine(z_vec, pk_vec, R_vec, S_vec, msg);
```

Reply:
Ok I like it!